chord/chordServerImpl.go

```diff
@@ -0,0 +1,68 @@
+package chord
+
+import (
+	"context"
+	"errors"
+	. "github.com/dosarudaniel/CS438_Project/services/chord_service"
+	"github.com/golang/protobuf/ptypes/empty"
+)
+
+// GetPredecessor (RPC) returns a pointer to the predecessor node
+func (chordNode *ChordNode) GetPredecessor(ctx context.Context, e *empty.Empty) (*Node, error) {
+	pred, doesExist := chordNode.getPredecessor()
+	if !doesExist {
+		return nil, errors.New("predecessor is nil")
+	}
+	return &pred, nil
+}
+
+// FIXME CheckPredecessor relies on use of ctx.Timeout
+/*
+ FindSuccessor finds the successor of id
+ n.find_successor(id)
+ 	if id is_in (n, successor]
+ 		return successor;
+ 	else
+ 		n' = closest_preceding_node(id);
+ 		return n'.find_successor(id);
+*/
+func (chordNode *ChordNode) FindSuccessor(ctx context.Context, messageIDPtr *ID) (*Node, error) {
+	if messageIDPtr == nil {
+		return nil, errors.New("id must not be nil")
+	}
+
+	id := messageIDPtr.Id
+	n := chordNode.node
+	succ, doesExist := chordNode.getSuccessor()
+	if !doesExist {
+		return nil, errors.New("successor does not exist")
+	}
+
+	if n.Id < id && id <= succ.Id {
```

Comment:
Same here, if `n.Id == 219` and `succ.Id == 33`   
for `id == 250` this will never be true, although it should.